algorithms/linfa-linear/Cargo.toml

```diff
@@ -17,18 +17,26 @@ keywords = ["machine-learning", "linfa", "ai", "ml", "linear"]
 categories = ["algorithms", "mathematics", "science"]
 
 [features]
-blas = ["ndarray-linalg", "linfa/ndarray-linalg", "argmin/ndarray-linalg"]
+blas = ["ndarray-linalg", "linfa/ndarray-linalg"]
+serde = ["serde_crate", "ndarray/serde", "argmin-math/ndarray_v0_15-serde" ]
+
+[dependencies.serde_crate]
+package = "serde"
+optional = true
+version = "1.0"
+default-features = false
+features = ["std", "derive"]
 
 [dependencies]
 ndarray = { version = "0.15", features = ["approx"] }
 linfa-linalg = { version = "0.1", default-features = false }
 ndarray-linalg = { version = "0.14", optional = true }
 num-traits = "0.2"
-argmin = { version = "0.4.6", features = ["ndarray", "ndarray-rand"] }
-serde = { version = "1.0", default-features = false, features = ["derive"] }
+argmin = { version = "0.6.0", default-features = false }
+argmin-math = { version = "0.1.0", features = ["ndarray_v0_15"] }
```

Comment:
Try `default-features = false, features = ["ndarray_0_15"]` to avoid pulling in `ndarray-linalg`. I'm not sure if you need the `primitive` feature as well.